src/EnliteMonolog/Service/MonologServiceFactory.php

```diff
@@ -130,8 +130,18 @@ public function createHandler($container, MonologOptions $options, $handler)
         }
 
         if (isset($handler['formatter'])) {
-            $formatter = $this->createFormatter($container, $handler['formatter']);
-            $instance->setFormatter($formatter);
+            if ($instance instanceof FormattableHandlerInterface
+                || !defined('Monolog\Logger::API')
+                || \Monolog\Logger::API === 1) {
+                $formatter = $this->createFormatter($container, $handler['formatter']);
+                $instance->setFormatter($formatter);
+            } else {
```

Comment:
it seems we're missing coverage over this branch. will you please add a test?

Reply:
apologies, i now see that `testCannotCreateWithFormatterWithoutCorrectInterface` covers this condition for `php:^7.2`.